### PR TITLE
Project-mongo-api | Week 14 | Petra Söderström 

### PR DESCRIPTION
**Pull request by Petrasoderstrom1612**

### CODE REVIEW

**Comment by Petrasoderstrom1612** [comment]
Cloud link to the project: https://project-mongo-api-bwwpzac7rq-lz.a.run.app/

File: server.js [hunk 1]
```diff
@@ -29,6 +71,85 @@ app.get("/", (req, res) => {
...
 
+// app.get("/songs", async (req, res) => {
+//   const allTheSongs = await Song.find({}) //to find all the songs
+//   res.status(200).json({
+//     success: true,
+//     body: allTheSongs
+//   });
+// });
+
+app.get("/songs/id/:id", async (req, res) => {
```
Comment (miadahlgren) [inline server.js:82]: Great work :) Maybe it had been more intuitive to have the endpoint /songs before songs/id/:id. But that is just a minor thing.

File: server.js [hunk 2]
```diff
@@ -29,6 +71,85 @@ app.get("/", (req, res) => {
...
+
+// app.get("/songs/genre/:genre/danceability/:danceability", async (req, res) 
+app.get("/songs/", async (req, res) => {
+  const {genre, danceability} = req.query /* req.params */; //deconstructing
+  const response = { //this line of code replaces the commented away snippets in try section
+    success: true,
+    body: {}
+  }
+
+  const matchAllRegex = new RegExp(".*");
```
Comment (miadahlgren) [inline server.js:117]: well done with the regex!

File: server.js [hunk 3]
```diff
@@ -29,6 +71,85 @@ app.get("/", (req, res) => {
...
+  const matchAllNumbersGreaterThanZero = { $gt: 0 };
```
Comment (miadahlgren) [inline server.js:118]: great job here. You really walked the extra mile :)

**Review by miadahlgren — COMMENTED** [review]
Great work Petra! Your code is well written and easy to follow. I appreciate that you added some extra stretch goals and seemed to have learnt a lot. :) The only thing i possibly could have a sked for is a list of endpoint on the first page, but that is just a minor thing! Well done!

**Pull request closed without merging**